beta.3/beta.4 removed public APIs: Model.build(), SequelizeConditions, generic <T> on findAll/findOne/find/count

## Summary

Starting from `2.15.0-beta.3`, several sequelize-compatibility APIs were removed from public type exports. These were available in `2.15.0-beta.2` and earlier. The removal makes it hard for downstream projects that rely on the sequelize adapter API surface to upgrade.

This is a follow-up of #467 (model compilation) and #475 / #476 (which were merged in beta.3 and are great — thanks!). The model compilation work itself is fine; the pain point is purely the removal of public types/methods that downstream code depends on.

## What was removed

### 1. `Model.build()` static method removed (TS2339)

`Bone.build(values)` / `SequelizeBone.build(values)` used to construct an **unsaved** model instance from a plain object. It is gone in beta.3+.

```ts
// worked in 2.15.0-beta.2
const post = Post.build({ title: 'Leah', authorId: 1 });
//                ^^^^^
// beta.3+: error TS2339: Property 'build' does not exist on type 'typeof Post'.
```

Real-world usage patterns that break:

```ts
// Rebuild an instance from raw row data (e.g. from a join result, a cached row, or a serialized payload)
const owner = User.build(row.dataValues);

// Build an instance to run through setters/encryption without persisting
const stub = Content.build({ body: plainBody });
stub.body; // triggers setter side-effects
```

`new Model(values)` is close, but it marks the instance as `isNewRecord = true`, so calling `.save()` later will `INSERT` instead of `UPDATE`. `build()` left `isNewRecord = false`, which is what we want when reconstructing rows that already exist in the DB.

### 2. `SequelizeConditions` / `SequelizeInstanceUpdateOptions` no longer exported (TS2305)

```ts
// worked in 2.15.0-beta.2
import {
  SequelizeBone,
  SequelizeConditions,
  SequelizeInstanceUpdateOptions,
} from 'leoric';

type FindOptions<T extends typeof SequelizeBone> = SequelizeConditions<T> & { chunk?: number };

async function safeUpdate<T extends typeof SequelizeBone>(
  this: T,
  values: Record<string, Literal>,
  opts: SequelizeInstanceUpdateOptions<this>,
) { /* ... */ }
```

```
beta.3+:
error TS2305: Module '"leoric"' has no exported member 'SequelizeConditions'.
error TS2305: Module '"leoric"' has no exported member 'SequelizeInstanceUpdateOptions'.
```

These types are the only way to type-hint `findAll` / `findOne` / `update` options in a way that matches the sequelize adapter's accepted shape (where/group/order/limit/offset/paranoid/...). Without them, downstream code has to either drop to `any` or re-declare the option shape by hand.

### 3. Generic type parameters dropped on `findAll` / `findOne` / `find` / `count` (TS2558)

```ts
// worked in 2.15.0-beta.2
const posts = await Post.findAll<Post>({ where: { ... } });
//                         ^^^^^
// beta.3+: error TS2558: Expected 0 type arguments, but got 1.
```

`Bone.findAll` / `findOne` / `find` / `count` / `aggregate` etc. no longer accept an explicit `<T>` type argument. The previous signatures were:

```ts
static findAll<T extends typeof SequelizeBone>(this: T, options?: SequelizeConditions<T>): Spell<T, Collection<InstanceType<T>>>;
static findOne<T extends typeof SequelizeBone>(this: T, options?: SequelizeConditions<T>): Spell<T, InstanceType<T>>;
static find<T extends typeof SequelizeBone>(this: T, ...): Spell<T, ...>;
static count<T extends typeof SequelizeBone>(this: T, ...): Spell<T, number>;
```

Removing `<T>` breaks any downstream code that passes the model subclass explicitly (common in generic helpers / mixins).

## Reproduction

```bash
# using 2.15.0-beta.2 — compiles clean
npm i leoric@2.15.0-beta.2
tsc --noEmit   # 0 errors

# using 2.15.0-beta.3 / beta.4 — same code fails
npm i leoric@2.15.0-beta.4
tsc --noEmit
# error TS2339: Property 'build' does not exist on type 'typeof Post'.
# error TS2305: Module '"leoric"' has no exported member 'SequelizeConditions'.
# error TS2558: Expected 0 type arguments, but got 1.
```

Minimal repro:

```ts
import { Bone, SequelizeBone, SequelizeConditions, DataTypes } from 'leoric';

class Post extends SequelizeBone {
  static table = 'posts';

  @Column(DataTypes.BIGINT)
  declare id: number;

  @Column()
  declare title: string;
}

// 1. build() removed
const p = Post.build({ title: 'Leah' });

// 2. SequelizeConditions removed
function find<T extends typeof SequelizeBone>(opts: SequelizeConditions<T>) {
  return T.findOne(opts);
}

// 3. generic arg removed
const list = await Post.findAll<Post>({ where: { id: 1 } });
```

## Impact on upgrade path

beta.3 / beta.4 contain fixes downstream users want (e.g. the `markModelClassReady` export in #476, the `Realm.query()` opts-forwarding fix in #475, the ES2022 model-compilation path in #467). But the three removals above force a large migration in any codebase that:

- Uses `Model.build()` to reconstruct rows from raw data (joins, cache, RPC payloads).
- Types query option shapes with `SequelizeConditions` / `SequelizeInstanceUpdateOptions`.
- Passes the model subclass explicitly as a generic arg to `findAll` / `findOne` / `find` / `count`.

## Request

Could these be re-added as deprecated (or kept) for the 2.15 line? Possible options:

1. **Re-export `SequelizeConditions` and `SequelizeInstanceUpdateOptions`** from `leoric` index (even if they now just alias `WhereConditions<T>` + a few extra fields). No runtime impact, pure type re-export.
2. **Restore `Bone.build(values, opts?)` static method** as a thin wrapper over `new Model(values, { isNewRecord: false, ...opts })`. This preserves the "unsaved instance with `isNewRecord = false`" semantics that callers rely on.
3. **Restore the generic `<T extends typeof SequelizeBone>` parameter** on `findAll` / `findOne` / `find` / `count` overloads. Even if `this: T` already infers `T`, allowing the explicit form is a strict superset and doesn't break inference.

If any of these are intentionally gone (e.g. `build` is replaced by something else, or `SequelizeConditions` was renamed), a pointer to the replacement would also work — we can do the migration.

Thanks!

--- a/package.json
+++ b/package.json
@@ -14,6 +14,7 @@
   },
   "exports": {
     ".": {
+      "types": "./lib/index.d.ts",
       "import": "./dist/index.js",
       "require": "./lib/index.js"
     }

